static/js/src/PurchaseModal/components/BuyButton/BuyButton.tsx

```diff
@@ -103,27 +107,37 @@ const BuyButton = ({ setError, quantity, product, action }: Props) => {
         },
         onError: (error) => {
           setIsLoading(false);
-          if (
-            error instanceof Error &&
-            error.message.includes("can only make one purchase at a time")
-          ) {
-            setError(
-              <>
-                You already have a pending purchase. Please go to{" "}
-                <a href="/account/payment-methods">payment methods</a> to retry.
-              </>
-            );
-          } else {
-            Sentry.captureException(error);
-            setError(
-              <>
-                Sorry, there was an unknown error with the payment. Check the
-                details and try again. Contact{" "}
-                <a href="https://ubuntu.com/contact-us">Canonical sales</a> if
-                the problem persists.
-              </>
-            );
-          }
+          console.log(error);
+          if (error instanceof Error)
+            if (
+              error.message.includes("can only make one purchase at a time")
+            ) {
+              setError(
+                <>
+                  You already have a pending purchase. Please go to{" "}
+                  <a href="/account/payment-methods">payment methods</a> to
+                  retry.
+                </>
+              );
+            } else if (error.message.includes("tax_id_invalid")) {
+              setFormikErrors({
+                VATNumber:
+                  "That VAT number is invalid. Check the number and try again.",
+              });
+              setError(
+                <>That VAT number is invalid. Check the number and try again.</>
+              );
```

Comment:
We also need to validate for another error code. Something along the lines of the following:

```
            } else if (error.message.includes("tax_id_cannot_be_validated")) {
              setFormikErrors({
                VATNumber:
                  "VAT number could not be validated at this time, please try again later or contact customer success if the problem persists.",
              });
              setError(
              <>
                VAT number could not be validated at this time, please try again
                later or contact
                <a href="mailto:customersuccess@canonical.com">
                  customer success
                </a>{" "}
                if the problem persists.
              </>              );
              ```